src/app/services/actions/snippet-action-creator.ts

```diff
@@ -0,0 +1,41 @@
+import { IAction } from '../../../types/action';
+import { IQuery } from '../../../types/query-runner';
+import { GET_SNIPPET_SUCCESS } from '../redux-constants';
+
+export function getSnippetSuccess(response: string): IAction {
+  return {
+    type: GET_SNIPPET_SUCCESS,
+    response,
+  };
+}
+
+export function getSnippet(language: string, sampleQuery: IQuery, dispatch: Function) {
+  const sample = { ...sampleQuery };
+
+  if (sample.sampleUrl) {
+    const pathname = new URL(sample.sampleUrl).pathname;
```

Comment:
If the query has query parameters the information is lost here and not forwarded to the snippet API to generate the appropriate snippet.